### PR TITLE
feat: Allow a single bulk-ingest per project only (DEV-3379)

### PR DESCRIPTION
**Pull request by seakayone**
- **Inline BulkIngestService trait**
- **feat: Allow a single bulk-ingest per project only**
- **feat: Prevent finalize to run if ingest is still in progress**

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]

**Comment by seakayone** [comment]
I am not entirely sure if `[429 Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429)` or `[409 Conflict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409)` would be most suited response status code.
wdyt?

**Comment by BalduinLandolt** [comment]
> I am not entirely sure if [429 Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429) or [409 Conflict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) would be most suited response status code.
> 
> Both 429 Too Many Requests and 409 Conflict can be suitable depending on the context and interpretation of our application's behavior.
> 
> 429 Too Many Requests: This status code indicates that the user has sent too many requests in a given amount of time. If we interpret the situation as the client trying to initiate too many bulk ingest requests simultaneously, 429 could be appropriate. However, it's important to note that this status code is typically used for rate limiting scenarios.
> 
> 409 Conflict: This status code is more generally used when there is a conflict with the current state of the server. If we see the situation as a conflict because there's already a bulk ingest process running and initiating another one would clash with that, 409 is suitable.
> 
> Given that we are concerned with concurrent bulk ingests, where initiating another bulk ingest while one is already running would be conflicting with the server's state, 409 Conflict seems more aligned with the situation. It explicitly communicates that there's a conflict with the current state, which is that a bulk ingest process is already underway.
> 
> wdyt?

I like `409 Conflict`. But ultimately, either of those is better than a broken ingest process. :)

**Review by mpro7 — APPROVED** [review]

**Review by siers — APPROVED** [review]
Cool!

File: src/main/scala/swiss/dasch/domain/BulkIngestService.scala [hunk 1]
```diff
@@ -40,13 +22,30 @@ object IngestResult {
...
+    key: ProjectShortcode,
+    zio: ProjectShortcode => IO[E, A],
+  ): IO[Option[Nothing], Fiber.Runtime[E, A]] = {
+    def getSemaphore =
+      semaphoresPerProject.getOrElseSTM(key, TSemaphore.make(1)).tap(semaphoresPerProject.put(key, _)).commit
+    def acquireWithTimeout(sem: TSemaphore) =
+      sem.acquire.as(sem).commit.timeout(Duration.fromMillis(400)).some
+    getSemaphore
+      .flatMap(acquireWithTimeout)
+      .flatMap(sem => zio.apply(key).logError.ensuring(sem.release.commit).forkDaemon)
```
Comment (siers) [inline src/main/scala/swiss/dasch/domain/BulkIngestService.scala:42]: The definitions here are a bit entangled, I suppose a for comprehension would've looked better here.

**Pull request merged**